File: muebleriaValadezBack/muebleriaValadezBack/Models/Sucursal.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace muebleriaValadezBack.Models
{
    public class Sucursal
    {
        [Key]
        public long IdSucursal { get; set; }

        [ForeignKey("direccion")]
        public long IdDireccion { get; set; }

        public Direccion direccion { get; set; }
        public String razonSocial { get; set; }


    }
}

File: muebleriaValadezFront/src/app/empleadoAdministrador/sucursales/sucursal.ts
export interface Sucursal {
    idSucursal: number;
    razonSocial: string;
    direccion: Direccion;
    estatus: string;
}

export interface Direccion {
    idDireccion: number;
    calle: string;
    noExt: number;
    noInt?: number;
    idDomicilio: number;
}